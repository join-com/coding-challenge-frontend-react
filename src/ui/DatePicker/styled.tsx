import styled from '@/libs/styled-components'

export const PickerContainer = styled.div`
  width: 200px;
  height: 100%;
  flex-grow: 0;
  flex-shrink: 0;
`
