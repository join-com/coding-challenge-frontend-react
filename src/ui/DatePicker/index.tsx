import React from 'react'
// @ts-ignore
import { DatePicker } from '@atlaskit/datetime-picker'
import { PickerContainer } from './styled'

export const FFDatePicker = ({
  input,
  placeholder,
  meta: { error, touched },
}: {
input: object
placeholder: string
meta: {
error?: string
touched?: boolean
}
}) => (
  <PickerContainer>
    <DatePicker
      {...input}
      placeholder={error && touched ? error : placeholder}
      isInvalid={error && touched}
    />
  </PickerContainer>
)
