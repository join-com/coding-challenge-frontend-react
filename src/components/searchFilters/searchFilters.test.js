import React from 'react';
import { shallow,mount } from 'enzyme';
import SearchFinters from './searchFilters';
import Grid from '@material-ui/core/Grid';
import DatePicker from "react-datepicker";
import DateRange from '@material-ui/icons/DateRange';
import SearchIcon from '@material-ui/icons/Search';

describe('Search Filters', () => {
    it('it should render correctly', () => {
        const component = shallow(<SearchFinters />);
        expect(component).toMatchSnapshot();
    });
    it('renders `datePicker',()=>{
        const component = mount(<SearchFinters />);
        expect(component.find('.datePicker').length).toBe(4)
    })
    it('renders `<Grid />',()=>{
        const component = mount(<SearchFinters />);
        expect(component.find(Grid).length).toBe(7)
    })
    it('renders `<DatePicker />',()=>{
        const component = mount(<SearchFinters />);
        expect(component.find(DatePicker).length).toBe(2)
    })
    it('renders `<DateRange />',()=>{
        const component = mount(<SearchFinters />);
        expect(component.find(DateRange).length).toBe(2)
    })
    it('renders `<SearchIcon />',()=>{
        const component = mount(<SearchFinters />);
        expect(component.find(SearchIcon).length).toBe(1)
    })

});