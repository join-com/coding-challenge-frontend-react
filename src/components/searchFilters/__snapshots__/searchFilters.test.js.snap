// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Filters it should render correctly 1`] = `
<SearchFilters
  classes={
    Object {
      "cardFooterContent": "SearchFilters-cardFooterContent-6",
      "dateRangeIcon": "SearchFilters-dateRangeIcon-5",
      "gridPad": "SearchFilters-gridPad-3",
      "gridPadInner": "SearchFilters-gridPadInner-4",
      "imgHover": "SearchFilters-imgHover-7",
      "main": "SearchFilters-main-1",
      "searchBtn": "SearchFilters-searchBtn-2",
      "searchIcon": "SearchFilters-searchIcon-8",
      "title": "SearchFilters-title-9",
    }
  }
/>
`;
